src/Compilers/CSharp/Test/Symbol/Symbols/LocalFunctionTests.cs

```diff
@@ -189,5 +189,181 @@ public void M()
+                    //     var y1 = new Func<string>(c.MyExtension);
+                    Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(10, 31),
+                    // (17,18): error CS8820: A static anonymous function cannot contain a reference to 'c'.
+                    //         var x2 = c.MyExtension;
+                    Diagnostic(ErrorCode.ERR_StaticAnonymousFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(17, 18),
+                    // (18,35): error CS8820: A static anonymous function cannot contain a reference to 'c'.
+                    //         var y2 = new Func<string>(c.MyExtension);
+                    Diagnostic(ErrorCode.ERR_StaticAnonymousFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(18, 35));
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup2()
+        {
+            CreateCompilation("""
+                using System;
+
+                var c = new C();
+                LocalFunc();
+                NonStatic();
+
+                static void LocalFunc()
+                {
+                    var x1 = Extensions.MyExtension;
+                    var y1 = new Func<C, string>(Extensions.MyExtension);
+                }
+
+                void NonStatic()
+                {
+                    Action f = static () =>
+                    {
+                        var x2 = Extensions.MyExtension;
+                        var y2 = new Func<C, string>(Extensions.MyExtension);
+                    };
+                }
+
+                public class C
+                {
+                }
+
+                public static class Extensions
+                {
+                    public static string MyExtension(this C c)
+                        => string.Empty;
+                }
+                """).VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup3()
+        {
+            CreateCompilation("""
+                using System;
+                public class Base { }
+
+                public class C : Base
+                {
+                    public void M()
+                    {
+                        LocalFunc();
+                        NonStatic();
+
+                        static void LocalFunc()
+                        {
+                            var x1 = this.MyExtension;
+                            var x2 = new Func<string>(this.MyExtension);
+                            var y1 = base.MyExtension;
+                            var y2 = new Func<string>(base.MyExtension);
+                        }
+                
+                        void NonStatic()
+                        {
+                            Action f = static () =>
+                            {
+                                var x3 = this.MyExtension;
```

Comment:
>this.MyExtension [](http://example.com/codeflow?start=41&length=16)

Should this case and the next be reported as errors?

Reply:
@cston I'm not sure, was already asking in https://github.com/dotnet/roslyn/pull/65965#discussion_r1053728427

Note that when I comment portion of the test and make the test input as follows:

```csharp
                using System;
                public class Base { }

                public class C : Base
                {
                    public void M()
                    {
                        //LocalFunc();
                        NonStatic();

                        //static void LocalFunc()
                        //{
                        //    var x1 = this.MyExtension;
                        //    var x2 = new Func<string>(this.MyExtension);
                        //    var y1 = base.MyExtension;
                        //    var y2 = new Func<string>(base.MyExtension);
                        //}
                
                        void NonStatic()
                        {
                            Action f = static () =>
                            {
                                var x3 = this.MyExtension;
                                var x4 = new Func<string>(this.MyExtension);
                                //var y3 = base.MyExtension;
                                //var y4 = new Func<string>(base.MyExtension);
                            };
                        }
                    }
                }

                internal static class Extensions
                {
                    public static string MyExtension(this Base c)
                        => string.Empty;
                }
```

errors are reported on x3 and x4 properly.

See also a simplified example that is unrelated to extension methods [here](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKAPQGYACLUgYQIG8DSHyz0AWUgWQAoBKexu/IyGkAcgHsYAZQAuEaVADGPANwE+w8m3FTZ8pb0EbSAo8PTZMpAGakAvOWwA2UjzsA+daeOevjAPR+pDBipACmAE7hYuHW0aQABtIAFlAAzvEAdKQAQgCu0qRQNvEAnvGFqaQKYmBgoTDSoQhwpBAwYZFxaaQADlEIuQpNPr6kAG4QMQAedqTJaRkA4qHSABIQqUmUYgihqoajjBMxJbPAG6EZ7PuHAL43RreeT/gvQA===)

Reply:
> Note that when I comment portion of the test and make the test input as follows:
> ...
> errors are reported on x3 and x4 properly.

In that case, let's add a test without the commented out parts, to verify those errors are reported when there are no other errors.